test/test_app.c

```diff
@@ -0,0 +1,555 @@
+/* Standard includes. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Ice incluudes. */
+#include "ice_api.h"
+#include "ice_data_types.h"
+#include "stun_serializer.h"
+
+typedef enum RequestType{
+    NOMINATING_CANDIDATE,
+    CONNECTIVITY_CHECK,
+    RESPONSE_FOR_REQUEST
+} RequestType_t;
+
+TransactionIdStore_t buffer[ICE_MAX_STORED_TRANSACTION_ID_COUNT] = { 0 };
+
+uint32_t computeCrc32( uint32_t initialResult, uint8_t * pBuffer, uint32_t bufferLength )
+{
+    return 5;
+}
+
+void computeHMAC( uint8_t * password, uint32_t passwordLength, uint8_t * pBuffer, uint32_t bufferLength, uint8_t * output, uint32_t * outputLength )
+{
+    *outputLength = 10;
+}
+
+/*--------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+void test_IceAgentInit( IceAgent_t * iceAgent )
+{
+    IceResult_t result;
+
+    char str1[] = "local", str2[] = "abc123", str3[] = "remote", str4[] = "xyz789";
+    char str5[] = "remote:local";
+
+    result = Ice_CreateIceAgent( iceAgent, str1, str2, str3, str4, str5, buffer, computeCrc32, computeHMAC );
+    
+    if( result == ICE_RESULT_OK )
+    {
+        printf("Creation of Ice Agent is successful.\n");
+    }
+    else
+    {
+        printf("Creation of Ice Agent failed.\n");
+    }
+}
+
+/*--------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+void test_GenerateHostCandidate( IceAgent_t * iceAgent )
+{
+    printf( "\nAdding Local Host candidates\n\n");
```

Comment:
Can we remove the debug left-overs

Reply:
If we really want an entry and exit print, let us have a macro like ENTRY() and EXIT() which is common to all functions.

Reply:
It is decided that the tests will be moved out to a different repo and we can update and add them back later.